### DISCUSSION TITLE
Supabase auth working across a multi-tenant app with dynamic subdomains

### DISCUSSION THREAD

**Discussion by moreofmorris** [supabase/supabase#19420] [Questions]
I've searched this board to help me answer this question but I can't find anything that quite seems to answer what I'm after. Also, I'm a designer first and though I can code, I won't be as advanced as many on this board.

I have a multi-tenant app that automatically generates subdomains for any project a user creates. Locally, at least, every time a new project is created it could be `http://summer-time.localhost:3000` or `http://new-world.localhost:3000`. Each of these projects have access control on them, depending on what the user has set up for them, so an outside user to the website could have access to one of these projects or two or three, it depends.

Ideally, I would have the same logged in session user across not only the base domain but any of the subdomains, too. This would seem to solve all my problems and keep things simple but this is where I'm struggling. Especially as access to these projects, on their subdomains, is done with the Magic Link `signInWithOtp()` method. So imagine an outside user wants access to one of these projects, they would enter their email in, get a magic link and then be directed back to the project subdomain with it all logged in nicely.

I've looked at the `config.toml` but I don't think I can set subdomain wildcards on either `site_url` or `additional_redirect_urls`, and when I'm calling `signInWithOtp()`, it won't let me set any of the dynamic subdomains there with it's `emailRedirectTo` option because I can't really list every single subdomain in the `config.toml` as these subdomains are dynamically created.

Can anyone please help? I'm tearing my hair out over this and may well need to abandon Supabase as I just can't work it out!

Thanks so, so much for any clues or helpful tips.

**Comment by GaryAustin1** [comment]
There are wildcards for redirectURLs.
https://supabase.com/docs/guides/auth/concepts/redirect-urls

Not sure that will solve all your issues, but should get you over that hurdle.

**Reply by moreofmorris** [reply]
Thanks so much for the reply. I've looked at this before but I don't think this applies to subdomains. I may well of missed something?

**Reply by GaryAustin1** [reply]
This seems to have more info:  https://github.com/orgs/supabase/discussions/5742

I find many threads that imply wildcards work anywhere, but I've not tested it.
https://github.com/supabase/supabase/issues/4808
https://twitter.com/supabase/status/1539561445414805506?lang=en

**Reply by moreofmorris** [reply]
Thanks again, Gary.

I was really encouraged when I saw the Twitter/X post you shared. So locally I have set that up in my `config.toml` file like this:

<img width="762" alt="CleanShot 2023-12-06 at 13 06 40@2x" src="https://github.com/supabase/supabase/assets/244516/bbc00af6-514e-4b98-9b91-5dbe88c50f20">

And was hoping that would work when it comes to how I've setup my subdomains. As mentioned, there are subdomains automatically generated so I can't really list them out, they have to be wildcards. So to see if this really would work, I'm even hardcoding the URL I want to redirect the user to from the magic link (just to see if it works, obviously not for production code):

<img width="547" alt="CleanShot 2023-12-06 at 13 07 22@2x" src="https://github.com/supabase/supabase/assets/244516/576af651-6d8e-4c9c-90a3-ee08922a675a">

But when it comes to seeing the magic link, it's still only going to `http://localhost:3000`:

![CleanShot 2023-12-06 at 13 00 37@2x](https://github.com/supabase/supabase/assets/244516/56bdf9b6-fce3-4947-8d24-55431b619267)

So I can't even seem to get that part of the puzzle working 🤔

**Reply by GaryAustin1** [reply]
If you hard code your last url in the .toml file does it work?

**Reply by moreofmorris** [reply]
So here is my `.toml` file:

<img width="778" alt="image" src="https://github.com/supabase/supabase/assets/244516/e55f5365-9edf-44ff-85ca-1c0621fc28f5">

And here is where I'm signing in:

<img width="560" alt="CleanShot 2023-12-06 at 14 32 37@2x" src="https://github.com/supabase/supabase/assets/244516/c37f62ed-1736-4d3c-91d0-464a3a9e43af">

And here is the magic link URL:

<img width="703" alt="CleanShot 2023-12-06 at 14 33 21@2x" src="https://github.com/supabase/supabase/assets/244516/12ad8420-1446-4279-818c-e753a33af7bc">

So even that doesn't seem to work, which now seems really odd! Why wouldn't that be working, I wonder?

**Reply by GaryAustin1** [reply]
Well that is good news for you as it is not a wild card issue.  I don't do self-hosted myself so not much help.

Your js call looks correct.  Maybe your .toml file is not being reloaded or that is not the correct format for redirectUrls (I have not looked).

**Comment by tberoiz** [comment]
Hey @moreofmorris, where you able to find a solution?
